### PR TITLE
Remove module cmake options

### LINKED ISSUE #5163: Remove module-specific compatibility and deprecation options
I propose that we remove the module-specific options `HPX_WITH_<modulename>_COMPATIBILITY_HEADERS`, `HPX_WITH_<modulename>_DEPRECATION_WARNINGS`, and `HPX_WITH_<modulename>_TESTS`. We can instead use the global `HPX_WITH_DEPRECATION_WARNINGS` and `HPX_WITH_TESTS`. I see no harm in leaving compatiblity headers until they are outright removed. I find the module-specific options unnecessarily fine-grained, and they add unnecessary noise (and options) to users configuring HPX. The initial idea was good but I don't think it works as well when we have close to 100 modules. Objections?

### PR DESCRIPTION
**Pull request by aurianer**
Replace module specific `HPX_<modulename>_WITH_DEPRECATION_WARNINGS` and `HPX_<modulename>_WITH_TESTS` with `HPX_WITH_DEPRECATION_WARNINGS` and `HPX_WITH_TESTS` respectively.
Fixes #5163

### CODE REVIEW

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

File: cmake/templates/compatibility_header.hpp.in [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include <hpx/@modulename@/config/defines.hpp>
 #include <@new_header@>
 
-#if HPX_@modulename_upper@_HAVE_DEPRECATION_WARNINGS
+#if HPX_HAVE_DEPRECATION_WARNINGS
 #if defined(HPX_MSVC)
 #pragma message("The header @old_header@ is deprecated, " \
     "please include @new_header@ instead")
```
Comment (msimberg) [inline cmake/templates/compatibility_header.hpp.in:18]: Should we add a message here telling users how to silence the warning, like @hkaiser has done for the `HPX_DEPRECATED_V*` macros?

**Review by msimberg — DISMISSED** [review]
Thanks @aurianer! Just one minor suggestion.

File: cmake/templates/compatibility_header.hpp.in [hunk 2]
```diff
@@ -15,9 +15,12 @@
...
 #pragma message("The header @old_header@ is deprecated, " \
-    "please include @new_header@ instead")
+    "please include @new_header@ instead." \
+    "To disable this warning, please define HPX_HAVE_DEPRECATION_WARNINGS=0.")
 #else
 #warning "The header @old_header@ is deprecated, " \
-    "please include @new_header@ instead"
+    "please include @new_header@ instead" \
+    "To disable this warning, please define HPX_HAVE_DEPRECATION_WARNINGS=0."
+
```
Comment (hkaiser) [inline cmake/templates/compatibility_header.hpp.in:24]: This is just a stray empty line here, I assume.

**Review by hkaiser — COMMENTED** [review]
Could you please update this as well: https://github.com/STEllAR-GROUP/hpx/blob/master/.cmake-format.py#L360?

**Pull request merged**